infrastructure/container_payload/run.py

```diff
@@ -174,113 +174,120 @@ def c(cl=[]):
             traceback.print_exc()
             return False, output_lines
 
+    def get_build_line(extra_args=[]):
+        line = ["gprbuild", "-q", "-P", "main", "-gnatwa"]
+        line.extend(extra_args)
+        return line
+
+    def build(extra_args=[]):
```

Comment:
it's dangerous for maintainibility to make a default assignments to a mutable type (https://docs.python-guide.org/writing/gotchas/) 

(same for the subprogram above).

This function needs a doctstring, especially since it returns something, which is not implied by its name...

Also, if get_build_line is only called by build (is this correct?), let's merge the two? (same for get_prove_line / prove and get_run_line / run)